### ISSUE TITLE
DRAFT: Multiple contact roles per ressource base

### ISSUE THREAD

**Issue by gannebamm** [GeoNode/geonode#10290]
**THIS IS A DRAFT - I WILL CONTINUE TO ELABORATE ON THIS LATER TODAY**

**Is your feature request related to a problem? Please describe.**
I want to use multiple CI CONTACT roles in the metadata of my datasets. 

a) I can not add multiple CI CONTACT roles to a dataset in the metadata wizard / advanced metadata editor.
b) I can not use all of the CI CONTACT roles ISO 19113 provides.
c) pycsw does not respond with all CI CONTACT roles given. It will only respond with POC ('pointOfContact'), the owner ('originator') and metadata author ('author'). 

I can put 1:N metadata roles in the Django admin area. Since Contact Roles are already implemented in the model:
https://github.com/GeoNode/geonode/blob/master/geonode/base/models.py#L105
which uses this enum with all ISO CI CONTACT roles: https://github.com/GeoNode/geonode/blob/6e093976049b6bff7d7af70f76df14e0f2f40e0a/geonode/people/enumerations.py#L21

but currently, you can only set POC and metadata author and owner by the metadata wizard. Some example of this is:
https://github.com/GeoNode/geonode/blob/6e093976049b6bff7d7af70f76df14e0f2f40e0a/geonode/base/models.py#L1850

**Describe the solution you'd like**
While POC, owner and metadata author are single contacts, ISO also provides roles which can have multiple persons involved.

**Describe alternatives you've considered**

**Additional context**

**Comment by mwallschlaeger** [comment]
Hey gannebamm, I'm also very interested in this feature. We're currently adopting the metadata schema for our geonode installation. I guess this changes we pretty much are similar to the ones we are planing. Would the other members of the SC accept this changes to the geonode core?

**Comment by t-book** [comment]
@mwallschlaeger My +1 ;)

**Comment by afabiani** [comment]
Just few headsup:
 - If you plan to touch the owner, this will imply a non-trivial work to be performed on security too (I would suggest to leave it as is)
 - The REST APIs must be updated too accordingly

**Comment by mwallschlaeger** [comment]
Thanks for your reply @afabiani and @t-book 

I also think keeping the owner as it is is a wise choice. I will come up with an PR as soon as I'm rdy.

**Comment by gannebamm** [comment]
@afabiani, since we will need to define third-party contact, do you think something like this #10342 is feasible?

**Comment by MalteIwanicki** [comment]
Hey @mwallschlaeger, did you start already with development on this? If so could you include me in the development process? I picked this issue up today and want to finish it in the upcoming two weeks.
Thanks

**Comment by mwallschlaeger** [comment]
@MalteIwanicki yes i'm on it. but it requires more investigation of code as i thought. we can discuss on working together on this. pls contact me via mail or glitter

**Comment by gannebamm** [comment]
The corresponding draft PR #10367 (which is not listed here for some reason)

**Comment by gannebamm** [comment]
@giohappy @afabiani 
Since the GeoNode 4.1 release was postponed but is likely imminent, shall we try to get this new feature into the upcoming 4.1 release?

**Comment by mwallschlaeger** [comment]
I will continue my work on the current change requests later this week, so you could have another look into this.

**Comment by giohappy** [comment]
@gannebamm there are plans to release a new version a few weeks after 4.1.0, that should include the refactored faceting/filtering system.
I think it will be safer to wait for that release, and take the time to test the multiple ISO roles better, unless you have urgencies.
